### PR TITLE
Installers: Add Base Installer Files Ahead Of JDK21 release.

### LINKED ISSUE #736: Add installer files for JDK21
Add installer files for JDK21

### PR DESCRIPTION
**Pull request by steelhead31**
DO NOT MERGE UNTIL CHECKSUMS ARE CORRECT  
  
Add the initial base files which will require the usual filename, checksum and version number updates during the release.

Fixes #736

### CODE REVIEW

**Comment by steelhead31** [comment]
Checks fail, as the binary files are not present yet.

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (adamfarley) [inline linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD:29]: Is the "U" correct, prior to October?

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (adamfarley) [inline linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD:36]: Should this be jdk20?

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (adamfarley) [inline linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD:98]: Another "U".

File: linux/jdk/debian/src/main/packaging/temurin/21/debian/rules
Comment (adamfarley) [inline linux/jdk/debian/src/main/packaging/temurin/21/debian/rules:6]: "U"

File: linux/jdk/debian/src/main/packaging/temurin/21/debian/rules
Comment (adamfarley) [inline linux/jdk/debian/src/main/packaging/temurin/21/debian/rules:8]: "U"

File: linux/jdk/suse/src/main/packaging/temurin/21/temurin-21-jdk.spec
Comment (adamfarley) [inline linux/jdk/suse/src/main/packaging/temurin/21/temurin-21-jdk.spec:100]: "U"

Also, I'm going to stop tagging all the "U"s. There's a bunch more of them, and the million one-letter-comments doesn't add value.

**Review by adamfarley — CHANGES_REQUESTED** [review]
A few points to consider. Once addressed (fixed/justified), will approve.

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (steelhead31) [reply linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD]: Yup, fixed!

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (steelhead31) [reply linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD]: Yup, fixed!

File: linux/jdk/debian/src/main/packaging/temurin/21/debian/rules
Comment (steelhead31) [reply linux/jdk/debian/src/main/packaging/temurin/21/debian/rules]: Yup, fixed!

File: linux/jdk/suse/src/main/packaging/temurin/21/temurin-21-jdk.spec
Comment (steelhead31) [reply linux/jdk/suse/src/main/packaging/temurin/21/temurin-21-jdk.spec]: Yup, fixed!

**Review by adamfarley — APPROVED** [review]
LGTM :)

**Review by karianna — CHANGES_REQUESTED** [review]

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (steelhead31) [reply linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD]: Fixed

File: linux/Jenkinsfile
Comment (steelhead31) [reply linux/Jenkinsfile]: Removed JDK20 :)

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (steelhead31) [reply linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD:10]: I think its correct, all of the previous versions of the APKBUILD file have the provider priority set as the JDK version, @gdams can you confirm ?

File: linux/jdk/debian/src/main/packaging/temurin/21/debian/rules
Comment (steelhead31) [reply linux/jdk/debian/src/main/packaging/temurin/21/debian/rules:4]: I've updated this to match the previous debian version, ie 1112 for JDK11, 1712 for JDK17 etc.

File: linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD
Comment (gdams) [reply linux/jdk/alpine/src/main/packaging/temurin/21/APKBUILD:10]: Yeah looking at how the community do this for Alpine we are doing the same which makes sense https://git.alpinelinux.org/aports/tree/community/openjdk17/APKBUILD?h=3.15-stable

**Review by karianna — APPROVED** [review]

**Comment by steelhead31** [comment]
DO NOT MERGE UNTIL CHECKSUMS ARE CORRECTED FOR RELEASE

File: linux/jdk/redhat/src/main/packaging/temurin/21/temurin-21-jdk.spec
Comment (steelhead31) [reply linux/jdk/redhat/src/main/packaging/temurin/21/temurin-21-jdk.spec]: Fixed, thanks!

**Comment by gdams** [comment]
@steelhead31 we've started publishing the binaries now. Can you update the shasums?

File: linux/jdk/debian/src/main/packaging/temurin/21/debian/control
Comment (gdams) [inline linux/jdk/debian/src/main/packaging/temurin/21/debian/control:8]:
```suggestion
Architecture: amd64 arm64 ppc64el
```

**Review by jiekang — APPROVED** [review]
Specs look fine to me. Would be nice if the GH check could test them. Wonder if it's worthwhile in making the specs usable when not all arches are available... but for a separate conversation

File: linux/jre/debian/src/main/packaging/temurin/21/debian/control
Comment (gdams) [inline linux/jre/debian/src/main/packaging/temurin/21/debian/control:8]:
```suggestion
Architecture: amd64 arm64 ppc64el
```

**Review by gdams — APPROVED** [review]
LGTM (subject to ppc64le either being removed or waiting until it's been shipped)

**Pull request merged**
